test/preview/components/builders/test_dynamic_prompt_builder.py

```diff
@@ -0,0 +1,139 @@
+from typing import List, Union
+
+import pytest
+from jinja2 import TemplateSyntaxError
+
+from haystack.preview.components.builders.dynamic_prompt_builder import DynamicPromptBuilder
+from haystack.preview.dataclasses import ChatMessage
+
+
+class TestDynamicPromptBuilder:
+    @pytest.mark.parametrize(
+        "expected_runtime_variables, chat_mode", [(["var1", "var2", "var3"], True), (["var1", "var2"], False)]
+    )
+    def test_initialization(self, expected_runtime_variables: List[str], chat_mode: bool):
+        builder = DynamicPromptBuilder(expected_runtime_variables, chat_mode)
+        assert builder.expected_runtime_variables == expected_runtime_variables
+        assert builder.chat_mode == chat_mode
+
+        # regardless of the chat mode
+        # we have inputs that contain: prompt_source, template_variables + expected_runtime_variables
+        expected_keys = set(expected_runtime_variables + ["prompt_source", "template_variables"])
+        assert set(builder.__canals_input__.keys()) == expected_keys
+
+        # response is always prompt regardless of chat mode
+        assert set(builder.__canals_output__.keys()) == {"prompt"}
+
+        # prompt_source is a list of ChatMessage or a string
+        assert builder.__canals_input__["prompt_source"].type == Union[List[ChatMessage], str]
+
+        # output is always prompt, but the type is different depending on the chat mode
+        if chat_mode:
+            assert builder.__canals_output__["prompt"].type == List[ChatMessage]
+        else:
+            assert builder.__canals_output__["prompt"].type == str
```

Comment:
Let's split this into two different tests, conditional in tests is most of the times a bad idea. 😬